packages/code-du-travail-ui/src/utils/flavors.ts

```diff
@@ -0,0 +1,95 @@
+//
+
+import { CSSProperties } from "react";
+
+//
+
+type Flavour =
+  | "primary"
+  | "secondary"
+  | "warning"
+  | "success"
+  | "info"
+  | "danger"
+  | "link"
+  | string;
+
+export const flavors: Flavour[] = [
+  "primary",
+  "secondary",
+  "warning",
+  "success",
+  "info",
+  "danger",
+  "link"
+];
+
+//
+
+/**
+ * Return the 1st truthy prop that is one of an Button versions
+ *
+ * TODO(douglasduteil): handle legacy potential bug
+ * * Here the function is return the "first" truthy prop however this prop my be
+ * * random as Objects have no key order in JS and it's using "Object.keys" to
+ * * list the props object
+ *
+ * @param props Props to look into
+ * @param prefix Props prefix
+ */
+export const getFirstFlavor = (
+  props: Partial<FlavourProps>,
+  prefix = "btn"
+) => {
+  const versionProps = (Object.keys(props) as Array<keyof FlavourProps>).find(
+    flavor => flavors.includes(flavor) && Boolean(props[flavor])
+  );
+  return (versionProps && `${prefix}__${versionProps}`) || "";
+};
+
+/**
+ * Removes falsy flavour values from the props
+ * @param props Props clean
+ */
+export const cleanProps: (
+  props: Partial<FlavourProps>
+) => Partial<FlavourProps> = props =>
+  (Object.keys(props) as Array<keyof FlavourProps>).reduce(
+    (newProps, key) => ({
+      ...newProps,
+      ...(flavors.includes(key) && Boolean(props[key]) === false
+        ? {}
+        : { [key]: String(props[key]) })
+    }),
+    {}
+  );
+
+//
+
+export interface FlavourProps {
```

Comment:
```
type Success = "success"
...
type FlavourProps = Primary | Secondary | Warning | Info | Success 
```
because `<Button primary secondary>hello</Button>` does'nt make sense.

Reply:
True :+1: